dd-sdk-android/src/main/kotlin/com/datadog/android/rum/gestures/DatadogGesturesListener.kt

```diff
@@ -0,0 +1,152 @@
+) :
+    GestureDetector.OnGestureListener {
+
+    // region GesturesListener
+
+    override fun onShowPress(e: MotionEvent) {
+        // No Op
+    }
+
+    override fun onSingleTapUp(e: MotionEvent): Boolean {
+        decorViewReference.get()?.let {
+            // we just intercept but not steal the event
+            val spanBuilder = rumTracer
+                .buildSpan(UI_TAP_ACTION_EVENT)
+            val target =
+                findTarget(
+                    it,
+                    e.x,
+                    e.y
+                )
+            spanBuilder.withTag(TAG_TARGET_CLASS_NAME, target.javaClass.canonicalName)
+            // we will handle general exceptions here. We don't know what might go wrong apart from
+            // ResourceNotFound exception.
+            @Suppress("TooGenericExceptionCaught")
+            try {
+                spanBuilder.withTag(
+                    TAG_TARGET_RESOURCE_ID,
+                    target.resources.getResourceEntryName(target.id)
+                )
+            } catch (e: Exception) {
+                sdkLogger.e("$TAG: Could not found resource name for target:${target.id}", e)
+            }
+            spanBuilder
+                .start()
+                .finish(DEFAULT_EVENT_DURATION)
+        }
+        return false
+    }
+
+    override fun onDown(e: MotionEvent): Boolean {
+        // No Op
+        // we just intercept but not steal the event
+        return false
+    }
+
+    override fun onFling(
+        e1: MotionEvent,
+        e2: MotionEvent,
+        velocityX: Float,
+        velocityY: Float
+    ): Boolean {
+        // No Op
+        // we just intercept but not steal the event
+        return false
+    }
+
+    override fun onScroll(
+        e1: MotionEvent,
+        e2: MotionEvent,
+        distanceX: Float,
+        distanceY: Float
+    ): Boolean {
+        // No Op
+        // we just intercept but not steal the event
+        return false
+    }
+
+    override fun onLongPress(e: MotionEvent) {
+        // No Op
+    }
+
+    // endregion
+
+    companion object {
+        private const val TAG = "DatadogGestureListener"
+        internal const val UI_TAP_ACTION_EVENT = "TapEvent"
+        internal const val TAG_TARGET_CLASS_NAME = "target.classname"
+        internal const val TAG_TARGET_RESOURCE_ID = "target.resourceId"
+        val DEFAULT_EVENT_DURATION = TimeUnit.MILLISECONDS.toMicros(16)
+
+        private fun findTarget(decorView: View, x: Float, y: Float): View {
+            val stack = LinkedList<View>()
+            stack.addFirst(decorView)
+            while (stack.isNotEmpty()) {
+                val view = stack.removeFirst()
+
+                if (isValidTarget(view)) {
+                    return view
+                }
+
+                if (view is ViewGroup) {
+                    handleViewGroup(view, x, y, stack)
+                }
+            }
+
+            return decorView
```

Comment:
Shouldn't we return `null` in this case (so marking this method to return optional)? I think of a case of full-screen non-interactive UI, such as blank screen with loading indicator and no call to action.

Reply:
the decorView should always be there...no matter what, we have actually a check before to see if this is null. In case no target was found the assumption is that is only the decorView there (transparent view) and therefore I return that as a target (I am not even sure if you can click on it) but in any case I just want to make sure I return something if no relevant clickable target was found.

Reply:
> In case no target was found the assumption is that is only the decorView there (transparent view) and therefore I return that as a target (I am not even sure if you can click on it) but in any case I just want to make sure I return something if no relevant clickable target was found.

Yes and that's the case - won't we then send "fake" spans for users tapping on the non-interactive screen? If there's no interaction on the screen, no user actions should be reported IMO 🤔.

Reply:
For what is worth, I agree with @ncreated here. Better report nothing than falsy data imho; but we should log cases when this happens.

Reply:
Hmmm....I am not sure if we should not send the spans....maybe the client wants to see if users are clicking on something that they should not click (unresponsive items or misleading UI).